### PR TITLE
New XML constants: PCMK_XA_REMOTE_NODE through PCMK_XA_STATE

### PR DESCRIPTION
**Pull request by nrwahl2**

### CODE REVIEW

File: lib/pengine/pe_output.c [hunk 1]
```diff
@@ -861,10 +861,14 @@ cluster_counts_xml(pcmk__output_t *out, va_list args) {
     int ndisabled = va_arg(args, int);
     int nblocked = va_arg(args, int);
 
-    xmlNodePtr nodes_node = pcmk__output_create_xml_node(out, "nodes_configured", NULL);
+    xmlNodePtr nodes_node = NULL;
     xmlNodePtr resources_node = pcmk__output_create_xml_node(out, "resources_configured", NULL);
+    char *s = NULL;
 
-    char *s = pcmk__itoa(nnodes);
+    nodes_node = pcmk__output_create_xml_node(out, PCMK_XE_NODES_CONFIGURED,
```
Comment (nrwahl2) [inline lib/pengine/pe_output.c:868]: Whoops, this changes the order of resources and nodes.

File: include/crm/common/resources.h [hunk 2]
```diff
@@ -378,8 +378,9 @@ typedef struct resource_object_functions_s {
      *
      * \return A node where the resource is active, preferring the source node
      *         if the resource is involved in a partial migration, or a clean,
-     *         online node if the resource's "requires" is "quorum" or
-     *         "nothing", otherwise NULL.
+     *         online node if the resource's \c PCMK_META_REQUIRES is
+     *         \c PCMK_VALUE_QUORUM or \c PCMK__VALUE_NOTHING, otherwise
+     *         \c NULL.
```
Comment (nrwahl2) [inline include/crm/common/resources.h]: This really belongs in the upcoming `PCMK_VALUE_QUORUM` commit. I can fix it (and will if there's other review), but it's only doxygen.

File: lib/common/ipc_attrd.c [hunk 3]
```diff
@@ -68,7 +68,8 @@ dispatch(pcmk_ipc_api_t *api, xmlNode *reply)
         pcmk__attrd_reply_unknown
     };
 
-    if (pcmk__str_eq((const char *) reply->name, "ack", pcmk__str_none)) {
+    if (pcmk__str_eq((const char *) reply->name, PCMK__XE_ACK,
+                     pcmk__str_none)) {
```
Comment (kgaillot) [inline lib/common/ipc_attrd.c:72]: This can use `pcmk__xe_is()`

File: lib/common/ipc_pacemakerd.c [hunk 4]
```diff
@@ -201,7 +201,8 @@ dispatch(pcmk_ipc_api_t *api, xmlNode *reply)
     const char *value = NULL;
     long long value_ll = 0;
 
-    if (pcmk__str_eq((const char *) reply->name, "ack", pcmk__str_none)) {
+    if (pcmk__str_eq((const char *) reply->name, PCMK__XE_ACK,
+                     pcmk__str_none)) {
```
Comment (kgaillot) [inline lib/common/ipc_pacemakerd.c:205]: ditto

File: lib/common/ipc_schedulerd.c [hunk 5]
```diff
@@ -84,7 +84,8 @@ dispatch(pcmk_ipc_api_t *api, xmlNode *reply)
     };
     const char *value = NULL;
 
-    if (pcmk__str_eq((const char *) reply->name, "ack", pcmk__str_casei)) {
+    if (pcmk__str_eq((const char *) reply->name, PCMK__XE_ACK,
+                     pcmk__str_none)) {
```
Comment (kgaillot) [inline lib/common/ipc_schedulerd.c:88]: ditto

**Review by kgaillot — COMMENTED** [review]
Looks good

File: lib/common/ipc_attrd.c (hunk 3 above)
Comment (nrwahl2) [reply lib/common/ipc_attrd.c]: Added commit on top

File: lib/pengine/unpack.c [hunk 6]
```diff
@@ -550,8 +550,7 @@ expand_remote_rsc_meta(xmlNode *xml_obj, xmlNode *parent, pcmk_scheduler_t *data
     for (attr_set = pcmk__xe_first_child(xml_obj); attr_set != NULL;
          attr_set = pcmk__xe_next(attr_set)) {
 
-        if (!pcmk__str_eq((const char *) attr_set->name,
-                          PCMK_XE_META_ATTRIBUTES, pcmk__str_casei)) {
```
Comment (nrwahl2) [inline lib/pengine/unpack.c:553]: This drops `pcmk__str_casei` but probably fine for an element name at least

Comment (kgaillot) [reply lib/pengine/unpack.c:553]: yep that's fine for element/attribute names (though not values, usually)

**Review by kgaillot — COMMENTED** [review]
might as well fix the two commit issues

**Comment by nrwahl2** [comment]
Whoops, forgot by the time I came back to it

**Comment by nrwahl2** [comment]
Those two should be fixed

**Pull request merged**